pkg/cmd/run.go

```diff
@@ -116,18 +117,40 @@ doppler run --command "YOUR_COMMAND && YOUR_OTHER_COMMAND"`,
 
 		secrets := getSecrets(localConfig, enableFallback, fallbackPath, fallbackReadonly, fallbackOnly, exitOnWriteFailure, passphrase)
 
+		if preserveEnv {
+			utils.LogWarning("Ignoring Doppler secrets already defined in the environment due to --preserve-env flag")
+		}
+
 		env := os.Environ()
+		existingEnvKeys := []string{}
```

Comment:
Very small nit, but we should use a map so that we don't have to loop over this array every time

Reply:
Whoops, good catch